rust/src/quic/logger.rs

```diff
@@ -0,0 +1,38 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::quic::QuicTransaction;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+
+fn log_template(tx: &QuicTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
+    js.set_uint("version", u32::from(tx.header.version.clone()).into())?;
```

Comment:
Why do we need clone and into ?
Could we not use ` as u64`  ?

Reply:
`clone` is to call the from implementation: `from(from: QuicVersion)`
`into()` calls the from implementation for u32 -> u64 in this case. I prefer to avoid using `as` to avoid accidental precision loss, both are equivalent here.